csFastFloat/FastDoubleParser.cs

```diff
@@ -19,22 +21,27 @@ namespace csFastFloat
 
   public static class FastDoubleParser
   {
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public  static double exact_power_of_ten(long power)
+    {
+#if NET5_0
+      Debug.Assert(power < Constants.powers_of_ten_double.Length);
+      ref double tableRef = ref MemoryMarshal.GetArrayDataReference(Constants.powers_of_ten_double);
+      return Unsafe.Add(ref tableRef, (IntPtr)power);
+#else
+      return Constants.powers_of_ten_double[power];
+#endif
 
-    public  static double exact_power_of_ten(long power) => Constants.powers_of_ten_double[power];
-
-
+    }
 
-    public static double ToFloat(bool negative, AdjustedMantissa am)
+    public static unsafe double ToFloat(bool negative, AdjustedMantissa am)
     {
       double d;
       ulong word = am.mantissa;
       word |= (ulong)(am.power2) << DoubleBinaryConstants.mantissa_explicit_bits;
       word = negative ? word | ((ulong)(1) << DoubleBinaryConstants.sign_index) : word;
 
-      unsafe
-      {
-        Buffer.MemoryCopy(&word, &d, sizeof(double), sizeof(double));
-      }
+      Unsafe.CopyBlockUnaligned(&d, &word, sizeof(double));
```

Comment:
`Buffer.MemoryCopy` does an argument-check, that will be dominant with the ptr-size copy. So this method is used, that doesn't perform argument validation.